File: Samples/BasicApplication/BasicApplication/BasicApplication.Script/Imports/FormContexts/Northwind.RegionForm.cs

namespace BasicApplication.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;
    using BasicApplication;

    public partial class RegionForm : PrefixedContext
    {
        public RegionForm(string idPrefix) : base(idPrefix) {}
    
        public IntegerEditor RegionID { get { return ById<IntegerEditor>("RegionID"); } }
        public StringEditor RegionDescription { get { return ById<StringEditor>("RegionDescription"); } }
    }
}

File: common-features/src/demo.northwind/Modules/ServerTypes/Demo/Northwind.RegionForm.ts
import { initFormType, IntegerEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface RegionForm {
    RegionID: IntegerEditor;
    RegionDescription: StringEditor;
}

export class RegionForm extends PrefixedContext {
    static readonly formKey = 'Northwind.Region';
    declare private static init: boolean;

    constructor(...args: ConstructorParameters<typeof PrefixedContext>) {
        super(...args);

        if (!RegionForm.init) {
            RegionForm.init = true;

            initFormType(RegionForm, [
                'RegionID', IntegerEditor,
                'RegionDescription', StringEditor
            ]);
        }
    }
}